src/main/java/org/folio/circulation/resources/RequestNoticeSender.java

```diff
@@ -248,27 +257,31 @@ private CompletableFuture<Result<Void>> sendNotice(Request request, UUID templat
 
     log.debug("sendNotice:: parameters request: {}, templateId: {}, eventType: {}",
       () -> request, () -> templateId, () -> eventType);
-    JsonObject noticeContext = createRequestNoticeContext(request);
-    NoticeLogContext noticeLogContext = NoticeLogContext.from(request)
-      .withTriggeringEvent(eventType.getRepresentation())
-      .withTemplateId(templateId.toString());
-    PatronNotice notice = buildEmail(request.getUserId(), templateId, noticeContext);
 
-    return patronNoticeService.sendNotice(notice, noticeLogContext);
+    return fetchAdditionalInfo(request)
+      .thenApply(r -> r.map(req -> {
+        JsonObject noticeContext = createRequestNoticeContext(req);
+        NoticeLogContext noticeLogContext = NoticeLogContext.from(req)
+          .withTriggeringEvent(eventType.getRepresentation())
+          .withTemplateId(templateId.toString());
+        PatronNotice notice = buildEmail(req.getUserId(), templateId, noticeContext);
+        return patronNoticeService.sendNotice(notice, noticeLogContext);
+      }))
+      .thenApply(r -> null);
```

Comment:
`.thenApply(r -> null)` replaces an instance of `Result` with null, and Result should never be null.

Please extract `req -> {...}` into a separate method.
```suggestion
      .thenCompose(r -> r.after(req -> {
        JsonObject noticeContext = createRequestNoticeContext(req);
        NoticeLogContext noticeLogContext = NoticeLogContext.from(req)
          .withTriggeringEvent(eventType.getRepresentation())
          .withTemplateId(templateId.toString());
        PatronNotice notice = buildEmail(req.getUserId(), templateId, noticeContext);
        return patronNoticeService.sendNotice(notice, noticeLogContext);
      }));
```